dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/persistence/file/batch/BatchFileOrchestratorTest.kt

```diff
@@ -0,0 +1,863 @@
+
+        // Then
+        assertThat(result).isNull()
+        verify(mockLogHandler).handleLog(
+            Log.ERROR,
+            BatchFileOrchestrator.ERROR_LARGE_DATA.format(Locale.US, dataSize, MAX_ITEM_SIZE)
+        )
+    }
+
+    @Test
+    fun `𝕄 create the rootDirectory 𝕎 getWritableFile() {root does not exist}`(
+        @IntForgery(min = 1, max = FileOrchestratorTest.MAX_LOG_SIZE) logSize: Int
+    ) {
+        // Given
+        fakeRootDir.deleteRecursively()
+
+        // When
+        testedOrchestrator.getWritableFile(logSize)
+
+        // Then
+        assertThat(fakeRootDir).exists().isDirectory()
+    }
+
+    @Test
+    fun `𝕄 delete obsolete files 𝕎 getWritableFile()`(
+        @IntForgery(min = 1, max = MAX_ITEM_SIZE) dataSize: Int,
+        @LongForgery(min = OLD_FILE_THRESHOLD, max = Int.MAX_VALUE.toLong()) oldFileAge: Long
+    ) {
+        // Given
+        assumeTrue(fakeRootDir.listFiles().isNullOrEmpty())
+        val oldTimestamp = System.currentTimeMillis() - oldFileAge
+        val oldFile = File(fakeRootDir, oldTimestamp.toString())
+        oldFile.createNewFile()
+        val youngTimestamp = System.currentTimeMillis() - RECENT_DELAY_MS - 1
+        val youngFile = File(fakeRootDir, youngTimestamp.toString())
+        youngFile.createNewFile()
+
+        // When
+        val start = System.currentTimeMillis()
+        val result = testedOrchestrator.getWritableFile(dataSize)
+        val end = System.currentTimeMillis()
+
+        // Then
+        checkNotNull(result)
+        assertThat(result)
+            .doesNotExist()
+            .hasParent(fakeRootDir)
+        assertThat(result.name.toLong())
+            .isBetween(start, end)
+        assertThat(oldFile).doesNotExist()
+        assertThat(youngFile).exists()
+    }
+
+    @Test
+    fun `𝕄 return new File 𝕎 getWritableFile() {no available file}`(
+        @IntForgery(min = 1, max = MAX_ITEM_SIZE) dataSize: Int
+    ) {
+        // Given
+        assumeTrue(fakeRootDir.listFiles().isNullOrEmpty())
+
+        // When
+        val start = System.currentTimeMillis()
+        val result = testedOrchestrator.getWritableFile(dataSize)
+        val end = System.currentTimeMillis()
+
+        // Then
+        checkNotNull(result)
+        assertThat(result)
+            .doesNotExist()
+            .hasParent(fakeRootDir)
+        assertThat(result.name.toLong())
+            .isBetween(start, end)
+    }
+
+    @Test
+    fun `𝕄 return existing File 𝕎 getWritableFile() {recent file exist with spare space}`(
+        @StringForgery(size = SMALL_ITEM_SIZE) previousData: String,
+        @IntForgery(min = 1, max = SMALL_ITEM_SIZE) dataSize: Int
+    ) {
+        // Given
+        assumeTrue(fakeRootDir.listFiles().isNullOrEmpty())
+        val previousFile = testedOrchestrator.getWritableFile(previousData.length)
+        checkNotNull(previousFile)
+        previousFile.writeText(previousData)
+        Thread.sleep(1)
```

Comment:
Because file names are based on the timestamp they're created at. Without this, the tests became flaky as sometimes they ran so fast that the newly created file had the exact same name as the `previousFile`.